common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -414,14 +415,16 @@ public EncryptionType getEncryptionType(@NonNull final String data) throws Unsup
         EncryptionType encryptionType = getEncryptionType(encryptedBlob);
 
         if (encryptionType == EncryptionType.USER_DEFINED) {
-            if (AuthenticationSettings.INSTANCE.getSecretKeyData() != null) {
+            if (ProcessUtil.isBrokerProcess(mContext)){
+                if (COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(packageName)) {
```

Comment:
There's some odd coupling in functionality here.  You probably want an else branch here to throw an IllegalStateException or panic or something for the case where the process name doesn't match the package name.

Reply:
Yeah, you know the package name.  If you pass that into your process util you won't need to look for both of them and have the confusing case where you found one in the package name and the other in the process name.

Reply:
Hm..  Either - or.  I see you're using it elsewhere, and I missed that on the first time through.